src/spock_recovery.c

```diff
@@ -0,0 +1,489 @@
+ *
+ * Removes the replication slot and frees associated resources.
+ * Used when a node is being decommissioned or recovery is no longer needed.
+ */
+void
+drop_recovery_slot(void)
+{
+	SpockRecoverySlotData *slot;
+	char		slot_name[NAMEDATALEN];
+
+	if (!SpockRecoveryCtx)
+		return;
+
+	LWLockAcquire(SpockRecoveryCtx->lock, LW_EXCLUSIVE);
+
+	slot = get_recovery_slot();
+
+	/* Nothing to do if slot doesn't exist */
+	if (!slot->active)
+	{
+		LWLockRelease(SpockRecoveryCtx->lock);
+		return;
+	}
+
+	/* Copy slot name for use outside the lock */
+	strncpy(slot_name, slot->slot_name, NAMEDATALEN - 1);
+	slot_name[NAMEDATALEN - 1] = '\0';
+
+	/* Reset slot to initial state in shared memory */
+	slot->slot_name[0] = '\0';
+	slot->restart_lsn = InvalidXLogRecPtr;
+	slot->confirmed_flush_lsn = InvalidXLogRecPtr;
+	slot->min_unacknowledged_ts = 0;
+	slot->active = false;
+	slot->in_recovery = false;
+
+	LWLockRelease(SpockRecoveryCtx->lock);
+
+	/* Drop the actual replication slot */
+	PG_TRY();
+	{
+		ReplicationSlotDrop(slot_name, true);
+		elog(LOG, "dropped recovery slot '%s'", slot_name);
+	}
+	PG_CATCH();
+	{
+		/* Log error but don't fail - slot might already be gone */
+		elog(WARNING, "failed to drop recovery slot '%s'", slot_name);
+	}
+	PG_END_TRY();
+}
+
+/*
+ * update_recovery_slot_progress
+ *
+ * Update the progress tracking information for a recovery slot.
+ *
+ * This function tracks the minimum unacknowledged timestamp and LSN positions
+ * across all peer nodes. Currently not fully implemented in minimal version.
+ *
+ * Args:
+ *		slot_name: Name of the slot to update
+ *		lsn: WAL position of the transaction
+ *		commit_ts: Commit timestamp of the transaction
+ *
+ * Note: This is a placeholder for future WAL advancement logic.
+ */
+void
+update_recovery_slot_progress(const char *slot_name, XLogRecPtr lsn,
+							   TimestampTz commit_ts)
+{
+	SpockRecoverySlotData *slot;
+
+	if (!SpockRecoveryCtx)
+		return;
+
+	if (!slot_name)
+		return;
+
+	LWLockAcquire(SpockRecoveryCtx->lock, LW_EXCLUSIVE);
+
+	slot = get_recovery_slot();
+
+	/* Only update if this is our slot and it's active */
+	if (slot->active && strcmp(slot->slot_name, slot_name) == 0)
+	{
+		/*
+		 * Track the oldest unacknowledged transaction timestamp.
+		 * This helps identify how far behind the slot might be.
+		 */
+		if (slot->min_unacknowledged_ts == 0 ||
+			commit_ts < slot->min_unacknowledged_ts)
+		{
+			slot->min_unacknowledged_ts = commit_ts;
+		}
+
+		/*
+		 * Track the confirmed flush LSN (latest position).
+		 * This shows the most recent transaction we've seen.
+		 */
+		if (slot->confirmed_flush_lsn == InvalidXLogRecPtr ||
+			lsn > slot->confirmed_flush_lsn)
+		{
+			slot->confirmed_flush_lsn = lsn;
+		}
+
+		/*
+		 * Track the restart LSN (earliest position needed).
+		 * This would be updated by WAL advancement logic.
+		 */
+		if (slot->restart_lsn == InvalidXLogRecPtr ||
+			lsn < slot->restart_lsn)
+		{
+			slot->restart_lsn = lsn;
```

Comment:
Assuming lsn is either commit lsn or end lsn here; Would it ever be less then restart_lsn?